modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -86,16 +98,17 @@ RadialReturnStressUpdate::calculateNumberSubsteps(const RankTwoTensor & strain_i
   return substep_number;
 }
 
+template <>
 void
-RadialReturnStressUpdate::updateState(RankTwoTensor & strain_increment,
-                                      RankTwoTensor & inelastic_strain_increment,
-                                      const RankTwoTensor & /*rotation_increment*/,
-                                      RankTwoTensor & stress_new,
-                                      const RankTwoTensor & /*stress_old*/,
-                                      const RankFourTensor & elasticity_tensor,
-                                      const RankTwoTensor & elastic_strain_old,
-                                      bool compute_full_tangent_operator,
-                                      RankFourTensor & tangent_operator)
+RadialReturnStressUpdateTempl<false>::updateState(RankTwoTensor & strain_increment,
```

Comment:
and here you add the other specialization with the error message. And you never implement the generic version of the template!

Reply:
> and here you add the other specialization with the error message. And you never implement the generic version of the template!

Thanks for the comments Daniel! Here is also where I am not very sure about.

The AD and non-AD versions of `updateState`, as well as `updateStateSubstep` require different numbers of inputs, therefore what I do is just to error out when the incorrect version of the functions are called...

Do you mean to have `RankTwoTensor`->`GenericRankTwoTensor<false>`, etc, for the non-AD version? I am not sure how that would benefit as only non-AD would call this function